### PR TITLE
Update Jackson to include CVE fixed in version 2.12.6

### PR DESCRIPTION
**Pull request by pratimsc**
### Type of change

Bugfix - Security Vulnerability: Strimzi Operator

- Bugfix

### Description

This fix includes changes to mitigate CVE reported at - https://github.com/FasterXML/jackson-databind/issues/3328

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: packaging/install/cluster-operator/042-Crd-strimzipodset.yaml [hunk 1]
```diff
@@ -68,9 +68,9 @@ spec:
                   matchLabels:
                     x-kubernetes-preserve-unknown-fields: true
                     type: object
-                description: Selector is a label query which matches all the pods
-                  managed by this `StrimziPodSet`. Only `matchLabels` is supported.
-                  If `matchExpressions` is set, it will be ignored.
+                description: "Selector is a label query which matches all the pods\
+                  \ managed by this `StrimziPodSet`. Only `matchLabels` is supported.\
+                  \ If `matchExpressions` is set, it will be ignored."
```
Comment (scholzj) [inline packaging/install/cluster-operator/042-Crd-strimzipodset.yaml:73]: I don't think this formatting should be here. It will probably cause some issues.

Comment (pratimsc) [reply packaging/install/cluster-operator/042-Crd-strimzipodset.yaml:73]: I have seen similar kind of formatting in yaml under `documentation/configuring/modules/install/cluster-operator/042-Crd-strimzipodset.yaml`

I can update the code base at:
```
public class CrdGenerator {
    public static final YAMLMapper YAML_MAPPER = new YAMLMapper()
```
To generate description as single, as found in the Helm chart: `helm-charts/helm3/strimzi-kafka-operator/crds/042-Crd-strimzipodset.yaml`

Is that acceptable @scholzj ?

Comment (scholzj) [reply packaging/install/cluster-operator/042-Crd-strimzipodset.yaml:73]: Yeah, basically, ideally, the YAMLs should not change just because we change Jackson version.

Comment (pratimsc) [reply packaging/install/cluster-operator/042-Crd-strimzipodset.yaml:73]: @scholzj  Looks like this pull request for handling null in YAML has introduced change in behaviour between version 2.11.x and 2.12.x of Jackson

https://github.com/FasterXML/jackson-dataformats-text/issues/253

I applying an explicit formatting in the Crd Generator, and pushing the changes. Hopefully, it will keep CRDs unchanged as Jackson versions get further updated.

**Comment by scholzj** [comment]
The tests seem to be passing ... I remembered what was the issue in the past: https://github.com/FasterXML/jackson-dataformats-text/issues/249 ... can you check if this works fine now?

**Comment by pratimsc** [comment]
> 

I updated the test to have multi line YAML value in `crd-generator/src/test/java/io/strimzi/crdgenerator/ExampleCrd.java` as :
```
@Crd.Spec.AdditionalPrinterColumn(
                name = "configYaml",
                description = "nest: This is a nested yaml\n" +
                        "lines:\n" +
                        "  - 2nd\n" +
                        "  - 3rd",
                jsonPath = "...",
                type = "string")
```
Successfully, passed test in local dev. 

- [ ]

File: docker-images/artifacts/kafka-thirdparty-libs/cc/pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     </licenses>
 
     <properties>
-        <cruise-control.version>2.5.82</cruise-control.version>
+        <cruise-control.version>2.5.85</cruise-control.version>
```
Comment (scholzj) [inline docker-images/artifacts/kafka-thirdparty-libs/cc/pom.xml:19]: The new test for the nexted YAMLs is great. But could you please leave the Cruise Control update for a separate PR?

Comment (pratimsc) [reply docker-images/artifacts/kafka-thirdparty-libs/cc/pom.xml]: Reverted it.

Comment (scholzj) [reply docker-images/artifacts/kafka-thirdparty-libs/cc/pom.xml]: Just to be clear - if you want to update Cruise Control it is fine -> just use separate PR ;-)

**Review by scholzj — APPROVED** [review]
LGTM. Great work. Thanks.

**Review by ppatierno — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.
